image/qcow2/qcow2.go

```diff
@@ -899,6 +899,10 @@ func (img *Qcow2) ReadAt(p []byte, off int64) (n int, err error) {
 	if len(p) == 0 {
 		return
 	}
+	if img.clusterSize == 0 {
+		err = fmt.Errorf("cluster size cannot be 0")
```

Comment:
Nit picking: since there are no arguments to be formatted use `errors.New` instead of `fmt.Errorf`:

```suggestion
		err = errors.New("cluster size cannot be 0")
```

Further nit-picking: Keep `img` validation checks together at the top (move this condition before the check for `len(p) == 0` check.